src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -560,23 +597,24 @@ private void GetItemsAndContinue(EvalResultDataItem dataItem, DkmWorkList workLi
             var numRows = rows.Count;
             Debug.Assert(count >= numRows);
             var results = new DkmEvaluationResult[numRows];
-            GetEvaluationResultsAndContinue(rows, results, 0, numRows, workList, inspectionContext, value.StackFrame,
-                () => ContinueWithExceptionHandling(
+            var wl = new WorkList(workList, e => completionRoutine(DkmEvaluationEnumAsyncResult.CreateErrorResult(e)));
+            GetEvaluationResultsAndContinue(rows, results, 0, numRows, wl, inspectionContext, value.StackFrame,
+                () => wl.ContinueWith(
                     () =>
                     {
                         completionRoutine(new DkmEvaluationEnumAsyncResult(results));
                         rows.Free();
-                    },
-                    e => completionRoutine(DkmEvaluationEnumAsyncResult.CreateErrorResult(e))));
+                    }));
+            wl.Execute();
         }
 
-        private void StoreResultAndContinue(DkmEvaluationResult result, ArrayBuilder<EvalResultDataItem> rows, DkmEvaluationResult[] results, int index, int numRows, DkmWorkList workList, DkmInspectionContext inspectionContext, DkmStackWalkFrame stackFrame, CompletionRoutine completionRoutine)
+        private void StoreResultAndContinue(DkmEvaluationResult result, ArrayBuilder<EvalResultDataItem> rows, DkmEvaluationResult[] results, int index, int numRows, WorkList workList, DkmInspectionContext inspectionContext, DkmStackWalkFrame stackFrame, CompletionRoutine completionRoutine)
         {
             results[index] = result;
             index++;
```

Comment:
Personally, I think it would be clearer to pass (index + 1) to the next call since no one else will consume the mutated state.